actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -80,7 +82,9 @@ def call(env)
           flag_cookies_as_secure! headers if @secure_cookies
         end
       else
-        return redirect_to_https request if @redirect
+        if @redirect && (@redirect[:constrain_to].nil? || @redirect[:constrain_to].call(request))
+          return redirect_to_https request
+        end
```

Comment:
Could remove the nil check here with `return redirect_to_https request if @constrain_to.call(request)` and adding `@constrain_to = @redirect && @redirect[:constrain_to] || proc { @redirect }` after `@redirect` has been assigned to something.